core/backend/src/IModelDb.ts

```diff
@@ -1429,25 +1431,55 @@ export abstract class IModelDb extends IModel {
   }
 
   /** Load all setting dictionaries in this iModel into `this.workspace.settings` */
-  private loadSettingDictionaries() {
+  private loadIModelSettings() {
     if (!this.nativeDb.isOpen())
       return;
 
     this.withSqliteStatement("SELECT Name,StrData FROM be_Prop WHERE Namespace=?", (stmt) => {
       stmt.bindString(1, IModelDb._settingPropNamespace);
       while (stmt.nextRow()) {
         try {
-          const dict = JSON.parse(stmt.getValueString(1));
-          this.workspace.settings.addDictionary(stmt.getValueString(0), SettingsPriority.iModel, dict);
+          const settings = JSON.parse(stmt.getValueString(1));
+          this.workspace.settings.addDictionary({ name: stmt.getValueString(0), priority: Settings.Priority.iModel }, settings);
         } catch (e) {
-          UnexpectedErrors.handle(e);
+          Workspace.exceptionDiagnosticFn(e as WorkspaceDb.LoadError);
         }
       }
     });
   }
 
+  /** @internal */
+  protected async loadWorkspaceSettings() {
```

Comment:
Whenever we open an IModel, we now see if any of its internal setting dictionaries has a setting called "settingsWorkspaces". If so we attempt to load settings from those WorkspaceDbs. Since this happens on "open", we don't throw any exceptions if there are problems and instead report those would-be exceptions to the host application (e.g. iTS) via the global variable `Workspace.exceptionDiagnosticFn`. I expect them to be shown in a diagnostics panel?